### PR TITLE
BROKERS: Storage Broker - Select Status Code By Code

### LINKED ISSUE #8: BROKERS: Storage Broker - Select Status Code By Code

### PR DESCRIPTION
**Pull request by cjdutoit**
Closes #8 

image

### CODE REVIEW

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    internal partial class StorageBroker
+    {
+        private IQueryable<StatusCode> statusCodes { get; set; }
+
+        public IQueryable<StatusCode> SelectAllStatusCodes()
+        {
+            return this.statusCodes;
+        }
+
+        public async ValueTask<StatusCode> SelectStatusCodeByCodeAsync(int code) =>
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:20]: if we don't have to await then we just return the value - no need for `ValueTask`

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Standard.REST.RESTFulSense.Brokers.Storages
+{
+    internal partial class StorageBroker
+    {
+        private IQueryable<StatusCode> statusCodes { get; set; }
+
+        public IQueryable<StatusCode> SelectAllStatusCodes()
+        {
+            return this.statusCodes;
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:17]: fat-arrow this one

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        public async ValueTask<StatusCode> SelectStatusCodeByCodeAsync(int code) =>
+            await Task.FromResult(this.statusCodes.FirstOrDefault(statusCode => statusCode.Code == code));
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:21]: `FirstOrDefault` is business logic

File: Standard.REST.RESTFulSense/Models/HttpStatusCodes/StatusCodes.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) 2023 - The Standard Community - All rights reserved.
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Models/HttpStatusCodes/StatusCodes.cs:2]: Extend the top and bottom lines to cover `All rights reserved.` also remove the year

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+using System.Linq;
+using Newtonsoft.Json;
+using Standard.REST.RESTFulSense.Models.HttpStatusCodes;
+
+namespace Standard.REST.RESTFulSense.Brokers.Storages
+{
+    internal partial class StorageBroker : IStorageBroker
+    {
+        public StorageBroker() =>
+            this.InitialiseStatusCodes();
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.cs:16]: Remove `this` it should exclusively be for private class members

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+using System.Linq;
+using System.Threading.Tasks;
+using Standard.REST.RESTFulSense.Models.HttpStatusCodes;
+
+namespace Standard.REST.RESTFulSense.Brokers.Storages
+{
+    internal partial class StorageBroker
+    {
+        private IQueryable<StatusCode> statusCodes { get; set; }
```
Comment (hassanhabib) [inline Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:13]: If this is a property then it should be `StatusCodes`.
If it's a private class member then it shouldn't have a `get;` and a `set;`

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs (hunk 3 above)
Comment (cjdutoit) [reply Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:21]: Ok, I will remove this method then and then we can filter in the foundation service.

File: Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs (hunk 6 above)
Comment (cjdutoit) [reply Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs:13]: As the property is private is in not correct starting with the lowercase s  => `statusCodes`

Comment (cjdutoit) [reply Standard.REST.RESTFulSense/Brokers/Storages/StorageBroker.StatusCodes.cs]: i found these 2 sites that documents the use of CamelCase public properties and camelCase for private properties.

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

**Pull request merged**